### PR TITLE
Add hard mode

### PR DESCRIPTION
**Pull request by neithernut**
This PR adds a "Hard mode" to be used in CI and (maybe) hooks.
Hard mode currently activates Werror, causing the compiler to abort compilation upon compiler warnings. Unused functions and arguments are excluded.

### CODE REVIEW

**Comment by neithernut** [comment]
Hard mode is activated by activation during a CMake run, e.g.

``` sh
cmake -DHARD_MODE=ON .
```

File: CMakeLists.txt [hunk 1]
```diff
@@ -7,6 +7,12 @@ project(libreset C)
 
 
 #
+# Project options
+#
+option(HARD_MODE "Enables extra checks ofr use during development" OFF)
```
Comment (matthiasbeyer) [inline CMakeLists.txt]: typo here.

Comment (neithernut) [inline CMakeLists.txt]: Fixed.

**Comment by neithernut** [comment]
Mh... "-Wno-error=unused" seems to be behave differently than expected. Working on a fix.

**Comment by neithernut** [comment]
Fixed.

**Comment by matthiasbeyer** [comment]
Looks good, feel free to clean up the history now.

**Comment by neithernut** [comment]
Done.

**Pull request merged**